src/main/java/ch/njol/skript/expressions/ExprBossBarFlags.java

```diff
@@ -0,0 +1,92 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.boss.BarFlag;
+import org.bukkit.boss.BossBar;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Name("Boss Bar Flags")
+@Description("The flags of a boss bar. These flags control the behavior of the boss bar.")
+@Examples({"add darken sky to flags of {_bar}t"})
+@Since("INSERT VERSION")
+public class ExprBossBarFlags extends PropertyExpression<BossBar, BarFlag> {
+
+	static {
+		register(ExprBossBarFlags.class, BarFlag.class, "flags", "bossbars");
+	}
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		this.setExpr((Expression<? extends BossBar>) exprs[0]);
+		return true;
+	}
+
+	@Override
+	protected BarFlag[] get(Event event, BossBar[] source) {
+		List<BarFlag> flags = new ArrayList<>();
```

Comment:
Should this be a set? This is a rare case where supporting multiple values is a bit odd.